addons/binding/org.openhab.binding.miele/src/main/java/org/openhab/binding/miele/internal/discovery/MieleMDNSDiscoveryParticipant.java

```diff
@@ -34,9 +35,10 @@
  * @author Karel Goderis - Initial contribution
  *
  */
+@Component(immediate = true)
```

Comment:
Do we need the service declaration here (`service = MDNSDiscoveryParticipant.class`)?

Reply:
The default works just fine when it's directly implementing interfaces:

```java
/**
 * The types under which to register this Component as a service.
 * 
 * <p>
 * If no service should be registered, the empty value
 * <code>&#x7B;&#x7D;</code> must be specified.
 * 
 * <p>
 * If not specified, the service types for this Component are all the
 * <i>directly</i> implemented interfaces of the class being annotated.
 * 
 * @see "The service element of a Component Description."
 */
Class<?>[] service() default {};
```

It's also used like that in most of the other bindings, see:
* [ChromecastDiscoveryParticipant](https://github.com/openhab/openhab2-addons/blob/f4c6646a35f8ea1be36f9f69f2bc6a4b39be6520/addons/binding/org.openhab.binding.chromecast/src/main/java/org/openhab/binding/chromecast/internal/ChromecastDiscoveryParticipant.java#L33-L34)
* [FreeboxServerDiscoveryParticipant](https://github.com/openhab/openhab2-addons/blob/f4c6646a35f8ea1be36f9f69f2bc6a4b39be6520/addons/binding/org.openhab.binding.freebox/src/main/java/org/openhab/binding/freebox/internal/discovery/FreeboxServerDiscoveryParticipant.java#L35-L36)

Reply:
I am okay with it. Just curious because I only have basic knowledge about OSGi.

But I think we are not strict with using it - or leaving it away. I found 11 implementations of `MDNSDiscoveryParticipant`; 5 of them use the declaration 6 do not:

- [NeeoBrainDiscovery](https://github.com/openhab/openhab2-addons/blob/f4c6646a35f8ea1be36f9f69f2bc6a4b39be6520/addons/binding/org.openhab.binding.neeo/src/main/java/org/openhab/binding/neeo/internal/discovery/NeeoBrainDiscovery.java#L42-L43)
- [HyperionDiscoveryParticipant](https://github.com/openhab/openhab2-addons/blob/f4c6646a35f8ea1be36f9f69f2bc6a4b39be6520/addons/binding/org.openhab.binding.hyperion/src/main/java/org/openhab/binding/hyperion/internal/HyperionDiscoveryParticipant.java#L33-L34)

Reply:
Yes we can do a follow up PR to make it more consistent. Removal of redundant properties makes the most sense IMHO. ESH has a lot more (non-binding) classes where the `service` property for  `@Component` annotated classes is omitted.

Reply:
Fine. Let's do it that way. No priority.